src/block_cache/block_chunkcache.c

```diff
@@ -28,28 +65,35 @@ static int
 __chunkcache_alloc(WT_SESSION_IMPL *session, WT_CHUNKCACHE_CHUNK *chunk)
 {
     WT_CHUNKCACHE *chunkcache;
-    WT_DECL_RET;
+    size_t bit_index;
+    uint8_t *map_byte;
 
     chunkcache = &S2C(session)->chunkcache;
+    bit_index = 0;
 
     if (chunkcache->type == WT_CHUNKCACHE_IN_VOLATILE_MEMORY)
-        ret = __wt_malloc(session, chunk->chunk_size, &chunk->chunk_memory);
+        WT_RET(__wt_malloc(session, chunk->chunk_size, &chunk->chunk_memory));
     else {
-#ifdef ENABLE_MEMKIND
-        chunk->chunk_memory = memkind_malloc(chunkcache->memkind, chunk->chunk_size);
+retry:
+        /* Use the bitmap to find the free chunks in the cache. */
+        WT_RET(__bitmap_find_free(session, &bit_index));
+
+        /* Mark the free chunk in the bitmap to in use. */
+        map_byte = (uint8_t *)&chunkcache->bitmap[bit_index / 8];
```

Comment:
Is there a need to type cast to `(uint8_t *)`?

Reply:
[690e97d](https://github.com/wiredtiger/wiredtiger/pull/9442/commits/690e97db34d0ac5088874a15cf526129f4065290)